db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/Predicate.java

```diff
@@ -1628,4 +1628,17 @@ public boolean isVarcharBinaryRelationalOperator() {
         return bron.getLeftOperand().getTypeServices().
                                      getTypeName().equals(TypeId.VARCHAR_NAME);
     }
+
+    @Override
+    public boolean hasCorrelatedSubquery() throws StandardException {
+        CollectNodesVisitor visitor= new CollectNodesVisitor(SubqueryNode.class);
```

Comment:
It may be expensive to scan the Predicate multiple times throughout join planning every time we check if a join is feasible.  I suggest we cache the result of this check in the Predicate object and re-use that result on subsequent calls to hasCorrelatedSubquery.

Reply:
Good point, I added caching logic.